backend/models/history.js

```diff
@@ -118,88 +93,92 @@ historySchema.statics.findByBranch = function(dbColOptions, branch, projection,
 	} else {
 		query.shared_id = stringToUUID(branch);
 	}
-	return History.findOne(
-		dbColOptions,
+
+	const sort = {sort: {timestamp: -1}};
+
+	const res = await findOne(
+		account,
+		model,
 		query,
-		projection,
-		{sort: {timestamp: -1}}
+		{...projection, ...sort}
 	);
+
+	return res;
 };
 
-historySchema.statics.revisionCount = async function(teamspace, model) {
+History.revisionCount = async function(teamspace, model) {
 	const query = {"incomplete": {"$exists": false}, "void": {"$ne": true}};
-	return (await find(teamspace, model, query)).count();
+	return await db.count(teamspace, getCollName(model), query);
 };
 
 // get the head of default branch (master)
-historySchema.statics.findLatest = function(dbColOptions, projection) {
-	return this.findByBranch(dbColOptions, null, projection);
+History.findLatest = async function(account, model, projection) {
+	return await this.findByBranch(account, model, null, projection);
 };
 
-historySchema.statics.findByUID = function(dbColOptions, revId, projection) {
+History.findByUID = async function(account, model, revId, projection) {
 	projection = projection || {};
-	return History.findOne(dbColOptions, { _id: stringToUUID(revId)}, projection);
-
+	return await findOne(account, model, { _id: stringToUUID(revId)}, projection);
 };
 
-historySchema.statics.updateRevision = async function(dbColOptions, modelId, data) {
-	if(utils.hasField(data, "void") &&
-		Object.prototype.toString.call(data.void) === "[object Boolean]") {
-		const rev = await History.findByUID(dbColOptions, modelId);
-		if(!rev) {
-			return Promise.reject(responseCodes.INVALID_TAG_NAME);
-		}
+History.updateRevision = async function(account, model, revId, voidValue) {
+	if(Object.prototype.toString.call(voidValue) === "[object Boolean]") {
+		voidValue = voidValue ? true : undefined;
+
+		const {result} = await db.update(account, getCollName(model), {_id: utils.stringToUUID(revId)} , { $set:  {void: voidValue}});
 
-		if(data.void) {
-			rev.void = true;
-		} else {
-			rev.void = undefined;
+		if(!result.n) {
+			throw responseCodes.INVALID_TAG_NAME;
 		}
 
-		return rev.save().then(() => responseCodes.OK);
+		return responseCodes.OK;
 
 	} else {
-		return Promise.reject(responseCodes.INVALID_ARGUMENTS);
+		throw responseCodes.INVALID_ARGUMENTS;
 	}
 };
 
-historySchema.statics.findByTag = function(dbColOptions, tag, projection) {
-	projection = projection || {};
-	return History.findOne(dbColOptions, { tag, incomplete: {"$exists": false }}, projection);
-};
+History.renameRevisionTag = async function(account, model, revId, tag) {
```

Comment:
Is this used anywhere? I don't think we allow users to change revision tags.... 😆 😟

Reply:
Is for doing this:
https://github.com/3drepo/3drepo.io/blob/99833d59dde119c13aa98784013932e68299ef52/backend/routes/history.js#L205-L223

Maybe the function name is wrong?

Reply:
I think this may be a legacy API that we don't use (and don't want to let user use). Unless you can see a reference to use this API on the front end we should remove it.

Reply:
I dont see any references in the frontend, I'll remove it.